Sources/LanguageServerProtocol/Types/LineIterator.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  LineIterator.swift
+//  LanguageServerProtocol
+//
+//  Created by Ryan Lovelett on 5/30/18.
+//
+
+private extension Character {
+
+    var isLineEnding: Bool {
+        return self == "\r\n" || self == "\r" || self == "\n"
+    }
+
+}
+
+struct Line {
+    let number: Int
+    let start: String.Index
+    let end: String.Index
+    let last: Bool
+
+    func contains(_ index: String.Index) -> Bool {
+        if last {
```

Comment:
Document why this condition is here. I know there is a failing test case for it.